src/chol_derivative.h

```diff
@@ -0,0 +1,183 @@
+#include <Rcpp.h>
+#define INCLUDE_RCPP
+#include "utils.h"
+#include "covariance.h"
+#include "tmb_includes.h"
+
+using namespace Rcpp;
+using std::string;
+
+struct chol {
```

Comment:
so in C++ here would be nice to have some comments on top of each class and function to explain why it is there and approximately what it is doing.